probeservices/checkin.go

```diff
@@ -0,0 +1,24 @@
+package probeservices
+
+import (
+	"context"
+
+	"github.com/ooni/probe-engine/model"
+)
+
+type checkInResult struct {
+	Tests model.CheckInInfo `json:"tests"`
+	V     int               `json:"v"`
+}
+
+// CheckIn pobes ask for tests to be run or otherwise go back to sleep
+// The config argument contains the mandatory settings.
+// Returns the list of tests to run and the URLs, on success, or an explanatory error, in case of failure.
+func (c Client) CheckIn(ctx context.Context, config model.CheckInConfig) (model.CheckInInfo, error) {
```

Comment:
```suggestion
func (c Client) CheckIn(ctx context.Context, config model.CheckInConfig) (*model.CheckInInfo, error) {
```

I think we want to return either a valid pointer or an error. This is a minor thing, but I actually happen to quite like this either-or semantics where we basically emulate the returning of a sum type.

After this change, tests need to be updated.